core/src/CMakeLists.txt

```diff
@@ -97,7 +96,10 @@ KOKKOS_LINK_TPL(kokkoscore PUBLIC LIBDL)
 KOKKOS_LINK_TPL(kokkoscore PUBLIC LIBRT)
 KOKKOS_LINK_TPL(kokkoscore PUBLIC PTHREAD)
 KOKKOS_LINK_TPL(kokkoscore PUBLIC ROCM)
-IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS)
-  KOKKOS_LINK_TPL(kokkoscore PUBLIC LIBATOMIC)
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS AND
+    CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
+  list(APPEND CMAKE_REQUIRED_LIBRARIES "atomic")
```

Comment:
Doesn't `CMAKE_REQUIRED_LIBRARIES` only have an effect on `CHECK_FUNCTION_EXISTS` and similar checking functions?

Reply:
i don't know but this seems to append it in the right place now ...

Reply:
I don't understand ether

Reply:
we can try I guess. I am no CMake expert ...